liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,410 @@
+        const tokensToDenom = exchangeRateMult.times(price.times(collateralMult));
+
+        if (supply[token][account]) {
+          // Supply balances are stored in cTokens and need to be multiplied by the
+          //   exchange rate, price and the collateral factor to determine the ETH value
+          //   of the collateral.
+          // Ref: https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Comptroller.sol#L754
+          supplyBalance = supplyBalance.plus(supply[token][account].times(tokensToDenom));
+        }
+
+        // Ref: https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Comptroller.sol#L757
+        if (borrow[token][account]) {
+          // Only need to multiply by the price.
+          borrowBalance = borrowBalance.plus(borrow[token][account].times(price));
+        }
+      });
+
+      // Remove non-borrowers
+      if (borrowBalance.eq(0)) {
+        delete accounts[account];
+      } else {
+        accounts[account].supplyBalance = supplyBalance;
+        accounts[account].borrowBalance = borrowBalance;
+        accounts[account].health = supplyBalance.dividedBy(borrowBalance);
+      }
+    });
+
+    // Check the Comptroller contract for actual liquidity if the health is below x
+    const lowHealthAccounts = [];
+    Object.keys(accounts).forEach((currentAccount) => {
+      if (accounts[currentAccount].health.isLessThan(data.lowHealthThreshold)) {
+        lowHealthAccounts.push(currentAccount);
+      }
+    });
+    await Promise.all(lowHealthAccounts.map(async (currentAccount) => {
+      const liquidity = await comptrollerContract.getAccountLiquidity(currentAccount);
+      const shortfallUSD = new BigNumber(ethers.utils.formatEther(liquidity[2]).toString());
+
+      // Health factor affects the liquidatable amount. Ex: Shortfall of $50 with a Health factor
+      // of 0.50 means that only $25 can be successfully liquidated. ( $25 supplied / $50 borrowed )
+      const liquidationAmount = shortfallUSD.multipliedBy(accounts[currentAccount].health);
+
+      // Create a finding if the liquidatable amount is below the threshold
+      // Shorten metadata to 2 decimal places
+      if (liquidationAmount.isGreaterThan(data.minimumLiquidationInUSD)) {
+        const newFinding = createAlert(
+          data.developerAbbreviation,
+          data.protocolName,
+          data.protocolAbbreviation,
+          data.alert.type,
+          data.alert.severity,
+          currentAccount,
+          liquidationAmount.dp(2),
+          shortfallUSD.dp(2),
+          accounts[currentAccount].health.dp(2),
+        );
+        findings.push(newFinding);
+      }
+      // Zero out the health on the low accounts so they may be re-scanned. (optional)
+      // accounts[currentAccount] = {};
```

Comment:
I can't quite determine whether this should be uncommented or not.  It seems like leaving it commented would result in not re-checking the token balances for accounts with a (currently) low health.  I will give more thought to uncommenting this, but perhaps you have a leaning one way or another.

Reply:
In the `lowHealthThreshold` setting, accounts under 1.03 health ratio will be checked for liquidity in the Comptroller contract. Uncommenting this line would also cause the script to re-check on chain for all account balances. Since the script is also watching for negative health actions such as borrows and withdraw collateral. Then it is not possible to for the balances to change undetected. 

The one reason for the rescan, is that the bot doesn't account for borrowing interest, so the longer the bot runs, the farther the data will skew. Accounts near the threshold can be rescanned for the most accurate balances. Or this can be done on some/all accounts every x blocks.

I'm leaning against using this rescan line. Current workarounds are to increase the threshold or restart the bot.

Reply:
Removed this entry in commit 17ed2a115215bd20ef5be24f23bd2a6ee856d910 . Also updated README to clarify the limitation.